### PR TITLE
Split install steps

### PR DESCRIPTION
**Pull request by christopherwharrop-noaa**
This pull request splits the Spack installation steps for Parsl and Flux into multiple Docker layers.  It also reworks how the Spack and Intel OneAPI get initialized in the container to avoid problems with Spack not setting the environment correctly.  Also, this PR reinstates docker layer caching in the github actions workflow and transitions to Docker Compose V2.

### CODE REVIEW

**Comment by christopherwharrop-noaa** [comment]
@NaureenBharwaniNOAA - I think I finally figured out how to deal with the mysterious Spack environment issues.  This is ready for a look.  I'll clean the commit message history up during the squash merge.

File: docker/master/Dockerfile [hunk 1]
```diff
@@ -3,8 +3,6 @@ FROM noaagsl/slurm-master:latest
...
     libpmi2-0-dev
 
-# Set up user shell init
-RUN echo "source /opt/intel/oneapi/setvars.sh > /dev/null 2>&1" >> /home/admin/.bash_profile \
- && echo ". /opt/spack/share/spack/setup-env.sh" >> /home/admin/.bash_profile \
- && echo "spack env activate flux" >> /home/admin/.bash_profile \
- && chown admin:admin /home/admin/.bash_profile
+# Set up Intel and Spack initialization
+RUN echo "source /opt/intel/oneapi/setvars.sh > /dev/null 2>&1" > /etc/profile.d/intel-oneapi.sh \
+ && echo "source /opt/etc/activate-parsl-flux.sh" > /etc/profile.d/parsl-flux.sh
```
Comment (NaureenBharwaniNOAA) [inline docker/master/Dockerfile:8]: The comment here says it's setup Spack initialization. This to me looks like it's activating the parsl-flux initialization? 

Should we be using `. /opt/spack/share/spack/setup-env.sh \` to set up the spack initialization? Do we need to update this comment?

File: docker/frontend/Dockerfile [hunk 2]
```diff
@@ -28,34 +28,45 @@ RUN apt-get update -y && apt-get install -y --no-install-recommends -o=Dpkg::Use
...
 
-# Set up user shell init
-RUN echo "source /opt/intel/oneapi/setvars.sh > /dev/null 2>&1" >> /home/admin/.bash_profile \
- && echo ". /opt/spack/share/spack/setup-env.sh" >> /home/admin/.bash_profile \
- && echo "spack env activate flux" >> /home/admin/.bash_profile \
- && chown admin:admin /home/admin/.bash_profile
+USER root
 
-SHELL ["/bin/bash"]
+# Set up Intel and Spack initialization
```
Comment (NaureenBharwaniNOAA) [inline docker/frontend/Dockerfile:68]: Same here for this comment

**Review by NaureenBharwaniNOAA — COMMENTED** [review]
Just a basic question on the comments

File: docker/master/Dockerfile (hunk 1 above)
Comment (christopherwharrop-noaa) [reply docker/master/Dockerfile:8]: Yes, that comment (and the others) could be more precise.

**Comment by christopherwharrop-noaa** [comment]
@NaureenBharwaniNOAA - Comments have been updated.

**Review by NaureenBharwaniNOAA — APPROVED** [review]
LGTM!

**Pull request merged**